music_assistant/providers/nugs/__init__.py

```diff
@@ -0,0 +1,481 @@
+            album.year = int(year)
+
+        return album
+
+    def _parse_playlist(self, playlist_obj):
+        """Parse nugs playlist object to generic layout."""
+        return Playlist(
+            item_id=playlist_obj["id"],
+            provider=self.lookup_key,
+            name=playlist_obj["name"],
+            provider_mappings={
+                ProviderMapping(
+                    item_id=playlist_obj["id"],
+                    provider_domain=self.domain,
+                    provider_instance=self.instance_id,
+                )
+            },
+            metadata=MediaItemMetadata(
+                images=[
+                    MediaItemImage(
+                        type=ImageType.THUMB,
+                        path=playlist_obj["imageUrl"],
+                        provider=self.lookup_key,
+                        remotely_accessible=True,
+                    )
+                ],
+            ),
+            is_editable=False,
+        )
+
+    def _parse_track(self, track_obj: dict, artist=None, album=None, image=None) -> Track:
+        """Parse response from inconsistent nugs.net APIs to a Track model object."""
+        track_id = (
+            track_obj.get("trackId") or track_obj.get("trackID") or track_obj.get("trackLabel")
+        )
+        track_name = track_obj.get("name") or track_obj.get("songTitle")
+
+        track = Track(
+            item_id=str(track_id),
+            provider=self.lookup_key,
+            name=track_name,
+            provider_mappings={
+                ProviderMapping(
+                    item_id=str(track_id),
+                    provider_domain=self.domain,
+                    provider_instance=self.instance_id,
+                    available=True,
+                )
+            },
+        )
+
+        if artist:
+            track.artists.append(artist)
+        if (
+            track_obj.get("artist")
+            and isinstance(track_obj.get("artist"), dict)
+            and track_obj["artist"].get("id")
+        ):
+            track.artists.append(
+                self._get_item_mapping(
+                    MediaType.ARTIST, track_obj["artist"]["id"], track_obj["artist"]["name"]
+                )
+            )
+        if not track.artists:
+            msg = "Track is missing artists"
+            raise InvalidDataError(msg)
+
+        if album:
+            track.album = album
+        if image is None and track_obj.get("image"):
+            image = track_obj["image"]["url"]
+        if image:
+            track.metadata.images = [
+                MediaItemImage(
+                    type=ImageType.THUMB,
+                    path=image,
+                    provider=self.lookup_key,
+                    remotely_accessible=True,
+                )
+            ]
+        duration = track_obj.get("durationSeconds") or track_obj.get("totalRunningTime")
+        if duration:
+            track.duration = int(duration)
+        return track
```

Comment:
So, there was something that just popped in my head. Which is the situation where a user is browsing an album tracks listing or playlist listing and then wants to add a track to the library. Can you test if that works without problems ? If not, we can consider implementing a basic (shortlived) cache here so we can return the cached copy of a parsed track in the "get_track" method. There is a chance this already works without problem. Anyways, its fine to fix later in a follow-up PR (if there even is an issue). 

Let's no longer hold up this PR and merge it so people can test it!

Reply:
When I attempt to add a track as a favorite I receive error `track://##### not found on provider nugs`

I'll be glad to work on a follow up PR to this one if you can point me in the right direction for implementing the cache